sqlair.go

```diff
@@ -0,0 +1,172 @@
+package sqlair
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/canonical/sqlair/internal/expr"
+)
+
+// Statement is a sql statement with valid sqlair DML.
+type Statement struct {
+	pe *expr.PreparedExpr
+}
+
+// Query stores a database query and iterates over the results.
+type Query struct {
+	pe  *expr.PreparedExpr
+	re  *expr.ResultExpr
+	err error
+	q   func() (*sql.Rows, error)
+}
+
+// Err returns the error, if any, that was encountered during iteration.
+// It should only be called after Close, or after Next has returned false (implicit close).
+func (q *Query) Err() error {
+	return q.err
+}
+
+type DB struct {
+	db *sql.DB
+}
+
+func NewDB(db *sql.DB) *DB {
+	return &DB{db: db}
+}
+
+// sqlDB returns the underlying database object.
+func (db *DB) sqlDB() *sql.DB {
+	return db.db
+}
+
+// Prepare checks that SQLair DML statements in a SQL query are well formed.
+// typeInstantiations must contain an instance of every type mentioned in the
+// SQLair DML of the query. These are used only for type information.
+func Prepare(query string, typeInstantiations ...any) (*Statement, error) {
```

Comment:
I'm not sure _checks_ is the right word here as it implies that the function returns a `bool`. I think instead, using some form of 'try' would fit better?
```
// Prepare tries to parse SQLair DML statements into SQL and extract type
// information from the passed instances of the types mentioned in query.
```
Also, is it possible to avoid 'DML'? I have a personal phobia of acronyms :P

Reply:
I've replaces "DML" with "expression" and revised the comment